pkg/subctl/cmd/create.go

```diff
@@ -0,0 +1,89 @@
+/*
+© 2020 Red Hat, Inc. and others.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"github.com/submariner-io/lighthouse/pkg/apis/lighthouse.submariner.io/v2alpha1"
+	lighthouseClientset "github.com/submariner-io/lighthouse/pkg/client/clientset/versioned"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+)
+
+var (
+	createCmd = &cobra.Command{
+		Use:   "create",
+		Short: "Create a submariner resource",
+		Long:  `This command provides an easy CLI to create specific submariner resources`,
+	}
+	createServiceExportCmd = &cobra.Command{
+		Use:   "service-export",
+		Short: "Creates ServiceExport resource",
+		Long:  `This command provides an easy CLI to create ServiceExport resource`,
+		Run:   createServiceExport,
+	}
+	serviceNamespace string
+)
+
+func init() {
+	addKubeconfigFlag(createCmd)
+	addServiceExportFlags(createServiceExportCmd)
+	createCmd.AddCommand(createServiceExportCmd)
+	rootCmd.AddCommand(createCmd)
+}
+
+func addServiceExportFlags(cmd *cobra.Command) {
+	cmd.Flags().StringVar(&serviceNamespace, "namespace", "default", "Namespace to create ServiceExport in")
+
+}
+func createServiceExport(cmd *cobra.Command, args []string) {
+	err := validateArguments(args)
+	exitOnError("Insufficient arguments", err)
+	config, err := getRestConfig(kubeConfig, kubeContext)
+	exitOnError("Error connecting to the target cluster", err)
+	_, clientSet, err := getClients(config)
+	exitOnError("Error connecting to the target cluster", err)
+	lhClientSet, err := createLighthouseClient(config)
+	exitOnError("Error connecting to the target cluster", err)
+	svcName := args[0]
+	_, err = clientSet.CoreV1().Services(serviceNamespace).Get(svcName, metav1.GetOptions{})
+	exitOnError(fmt.Sprintf("Unable to find service %s in namespace %s", svcName, serviceNamespace), err)
```

Comment:
Users might want to export services before they’re created, shouldn’t we allow that? Technically, the order doesn’t matter, does it?

Reply:
> Users might want to export services before they’re created, shouldn’t we allow that? Technically, the order doesn’t matter, does it?

Currently we don't, adding this support in 0.5.0. Once we add, will remove this check.